### ISSUE TITLE
Admin\Facilities - External User can edit a facility

### ISSUE THREAD

**Issue by zoeyli-46** [bcgov/cas-registration#1608]
#### Description:
As an External User, I can edit my facility details
[Figma link ](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=12902-157&t=gXGIuv0Zz0q8D4AX-4)

#### Acceptance Criteria:

Given I am an authenticated user with the `industry-user` or `industry-user-admin` role,
When I am on the facilities page
Then I can click view details, for each row of facilities

Given that I am in the facilities details page,
When I click edit
Then the fields turn editable

Given that I am in edit mode
When I click submit
Then I get a confirmation that the edits have been made successfully

#### Development Checklist:

Refactor shared `route` component:
- [x] Move and rename `bciers/apps/administration/app/components/routes/facilities/Page.tsx` to `bciers/apps/administration/app/components/facilities/FacilitiesPage.tsx`
- [x] Delete `bciers/apps/administration/app/components/routes/facilities/Page.tsx` 
- [x] In`bciers/apps/administration/app/bceidbusiness/industry_user_admin/operations/[operationId]/facilities/page.tsx` change `import FacilitiesPage from "@/administration/app/components/routes/facilities/Page";` to `import FacilitiesPage from "@/administration/app/components/facilities/FacilitiesPage"`
- [x] In`bciers/apps/administration/app/bceidbusiness/industry_user/operations/[operationId]/facilities/page.tsx` change `import FacilitiesPage from "@/administration/app/components/routes/facilities/Page";` to `import FacilitiesPage from "@/administration/app/components/facilities/FacilitiesPage"`
- [x] In `bciers/apps/administration/app/idir/cas_admin/operations/[operationId]/facilities/page.tsx` change `import FacilitiesPage from "@/administration/app/components/routes/facilities/Page";` to `import FacilitiesPage from "@/administration/app/components/facilities/FacilitiesPage"`
- [x] In `bciers/apps/administration/app/idir/cas_analyst/operations/[operationId]/facilities/page.tsx` change `import FacilitiesPage from "@/administration/app/components/routes/facilities/Page";` to `import FacilitiesPage from "@/administration/app/components/facilities/FacilitiesPage"`
- [x] In `bciers/apps/administration/app/components/facilities/FacilitiesPage.tsx` change `import { FacilitiesSearchParams } from "../../facilities/types";` to `import { FacilitiesSearchParams } from "../facilities/types";`
- [x] In  `bciers/apps/administration/app/components/facilities/FacilitiesPage.tsx` change `import Facilities from "../../facilities/Facilities";` to `import Facilities from "../facilities/Facilities";`

Refactor `ActionCellFactory.`  to handle dynamic href property
- [x] Move `bciers/apps/administration/app/components/datagrid/ActionCellFactory.tsx` to `bciers/libs/components/src/datagrid/cells/ActionCellFactory.tsx` 
- [x] Modify `bciers/libs/components/src/datagrid/cells/ActionCellFactory.tsx`  to accept a function that generates the href dynamically based on the cell parameters
```
import Link from "next/link";
import { GridRenderCellParams } from "@mui/x-data-grid"; // or wherever GridRenderCellParams is imported from

const ActionCellFactory = ({
  getHref,
  replace,
  className,
  cellText,
}: {
  getHref: (params: GridRenderCellParams) => string;
  replace: boolean;
  className: string;
  cellText: string;
}) => {
  return (params: GridRenderCellParams) => {
    const href = getHref(params);
    return (
      <div>
        <Link className={className} href={href} replace={replace}>
          {cellText}
        </Link>
      </div>
    );
  };
};

export default ActionCellFactory;

```
- [x] In `bciers/apps/administration/app/components/facilities/FacilityDataGrid.tsx`  change to
```
"use client";

import { useMemo, useState } from "react";
import DataGrid from "@bciers/components/datagrid/DataGrid";
import facilityColumns from "../datagrid/models/facilities/facilityColumns";
import facilityGroupColumns from "../datagrid/models/facilities/facilityGroupColumns";
import { FacilityRow } from "./types";
import createFetchFacilitiesPageData from "./createFetchFacilitiesPageData";
import HeaderSearchCell from "@bciers/components/datagrid/cells/HeaderSearchCell";
import ActionCellFactory from "@bciers/components/datagrid/cells/ActionCellFactory";

const FacilityDataGrid = ({
  operationId,
  initialData,
}: {
  operationId: string;
  initialData: {
    rows: FacilityRow[];
    row_count: number;
  };
}) => {
  const createFacilitiesActionCell = () =>
    ActionCellFactory({
      getHref: (params) =>
        `/operations/${operationId}/facilities/${params.row.id}?title=${params.row.name}`,
      replace: true,
      className: "no-underline text-bc-link-blue whitespace-normal",
      cellText: "View Details",
    });

  const ActionCell = useMemo(() => createFacilitiesActionCell(), []);
  const [lastFocusedField, setLastFocusedField] = useState<string | null>(null);

  const SearchCell = useMemo(
    () => HeaderSearchCell({ lastFocusedField, setLastFocusedField }),
    [lastFocusedField, setLastFocusedField],
  );

  const columns = useMemo(() => facilityColumns(ActionCell), [ActionCell]);

  const columnGroup = useMemo(
    () => facilityGroupColumns(SearchCell),
    [SearchCell],
  );

  return (
    <DataGrid
      columns={columns}
      columnGroupModel={columnGroup}
      fetchPageData={createFetchFacilitiesPageData(operationId)}
      paginationMode="server"
      initialData={initialData}
    />
  );
};

export default FacilityDataGrid;

```
- [x] In bciers/apps/administration/app/components/facilities/FacilitiesForm.tsx` remove isCreating as param to SingleStepTaskListForm`
- [x] Create API service in `bc_obps/service/facility_service.py`
- [x] Create API data access service in `bc_obps/service/data_access_service/facility_service.py`
- [x] Create API endpoint in `bc_obps/registration/api/v1/_facilities/facility_id.py`
- [x] Create API test of service in `bc_obps/service/tests/test_facility_service.py`
- [x] Create API test of endpoint in `bc_obps/registration/tests/endpoints/v1/_facilities/test_facility_id.py`
- [x] Run python tests `cd bc_obps && make pythontests`
- [x] Run vitests `cd bciers &&yarn admin:test`

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Dependencies
Blocked by #1361

**Comment by shon-button** [comment]
@patriciarussellCAS

Reachable at:
https://cas-bciers-frontend-dev.apps.silver.devops.gov.bc.ca/administration/operations/002d5a9e-32a6-4191-938c-2c02bfec592d/facilities?operations_title=Operation+2

**Comment by patriciarussellCAS** [comment]
Review notes: 
 When I am on the Operations table page and I click on "Edit details" in the facility column, I am sent directly to the "add facility" page for a specific facility. Is this the intention? 
I would have though I would first get sent to the page with the list of all facilities for that operation, so I could decided which facility information page I want to edit.
